index.md

```diff
@@ -38,122 +70,646 @@ Below are the __four scrum ceremonies__ that occur in each sprint.
+### Security Assertion Markup Language (SAML)
+
+### Serverless
+
+Serverless is a cloud-native development model that allows developers to build and run applications without having to manage servers. There are still servers in serverless, but they are abstracted away from app development.
+
+### Setter Injection
+
+Spring Setter Injection is nothing but injecting the Bean Dependencies using the Setter methods on an Object. Unlike Spring Constructor Injection, in Setter Injection, the object is created first and then the dependency is injected.
+
+### Singleton Pattern
+
+Java Singleton Pattern is one of the Gangs of Four Design patterns and comes in the Creational Design Pattern category. Singleton pattern restricts the instantiation of a class and ensures that only one instance of the class exists in the Java virtual machine. It seems to be a very simple design pattern but when it comes to implementation, it comes with a lot of implementation concerns.
+
+Singleton pattern restricts the instantiation of a class and ensures that only one instance of the class exists in the java virtual machine.
+The singleton class must provide a global access point to get the instance of the class.
+Singleton pattern is used for logging, drivers objects, caching and thread pool.
+Singleton design pattern is also used in other design patterns like Abstract Factory, Builder, Prototype, Facade etc.
+Singleton design pattern is used in core java classes also, for example java.lang.Runtime, java.awt.Desktop.
+
+### Site Reliability Engineering (SRE)
+
+Site reliability engineering (SRE) is a set of principles and practices that incorporates aspects of software engineering and applies them to infrastructure and operations problems.
+The main goals are to create scalable and highly reliable software systems. Site reliability engineering is closely related to DevOps, a set of practices that combine software development and IT operations, and SRE has also been described as a specific implementation of DevOps.
+
+### SOAP
+
+SOAP stands for Simple Object Access Protocol. This method of web services transmits messages via XML, usually as an HTTP POST. SOAP services contain service operations that are requested by the calling application. These service operations usually contain verb/noun combinations (i.e. getAccountByID, getAccountByName, UpdateAccount…). One advantage to SOAP is that it provides WSDL (Web Service Definition Language) that describes the service operations. Most large enterprise applications support SOAP.
+
+### Spring Framework
 
-__Serverless__: 
+Spring was built on top of the idea of dependency injection and inversion of control. In normal words - instead of having a bunch of classes creating each other and passing each other from one place to another. You have a bag of beans, each bean declares its dependencies (what services do I need to work?) and Spring container resolves this requirements by automatically wiring everything together.
```

Comment:
Patterns & Technologies that Spring Framework has - Inversion of Control, Aspected Oriented Programming, Dependency Injection, Spring Data, Spring Cloud..